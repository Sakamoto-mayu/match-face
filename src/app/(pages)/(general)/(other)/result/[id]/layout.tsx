export const metadata = {
  title: "回答確認",
  description: "一般ユーザー回答確認ページ",
};

export default function ResultLayout({
  children,
}: {
  children: React.ReactNode;
}) {
  return <>{children}</>;
}
