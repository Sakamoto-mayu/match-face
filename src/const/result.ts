export const Answer = [
  {
    user_id: "user1",
    project_id: "1",
    question_id: "1",
    answer_date: new Date("2023-07-06"),
    content:
      "回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。",
  },
  {
    user_id: "user1",
    project_id: "1",
    question_id: "2",
    answer_date: new Date("2023-07-06"),
    content: "選択肢A",
  },
  {
    user_id: "user1",
    project_id: "1",
    question_id: "3",
    answer_date: new Date("2023-07-06"),
    content:
      "回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。",
  },
  {
    user_id: "user1",
    project_id: "2",
    question_id: "1",
    answer_date: new Date("2023-07-06"),
    content:
      "回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。",
  },
  {
    user_id: "user1",
    project_id: "2",
    question_id: "2",
    answer_date: new Date("2023-07-06"),
    content:
      "回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。",
  },
  {
    user_id: "user1",
    project_id: "2",
    question_id: "3",
    answer_date: new Date("2023-07-06"),
    content:
      "回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。",
  },
  {
    user_id: "user1",
    project_id: "3",
    question_id: "1",
    answer_date: new Date("2023-07-06"),
    content:
      "回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。",
  },
  {
    user_id: "user1",
    project_id: "3",
    question_id: "2",
    answer_date: new Date("2023-07-06"),
    content:
      "回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。",
  },
  {
    user_id: "user1",
    project_id: "3",
    question_id: "3",
    answer_date: new Date("2023-07-06"),
    content:
      "回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。",
  },
];

export const ModelAnswer = [
  {
    editor_id: "admin1",
    project_id: "1",
    question_id: "1",
    edit_date: new Date("2023-07-07"),
    content:
      "模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。",
  },
  {
    editor_id: "admin1",
    project_id: "1",
    question_id: "2",
    edit_date: new Date("2023-07-07"),
    content:
      "模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。",
  },
  {
    editor_id: "admin1",
    project_id: "1",
    question_id: "3",
    edit_date: new Date("2023-07-07"),
    content:
      "模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。",
  },
  {
    editor_id: "admin1",
    project_id: "2",
    question_id: "1",
    edit_date: new Date("2023-07-07"),
    content:
      "模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。",
  },
  {
    editor_id: "admin1",
    project_id: "2",
    question_id: "2",
    edit_date: new Date("2023-07-07"),
    content:
      "模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。",
  },
  {
    editor_id: "admin1",
    project_id: "2",
    question_id: "3",
    edit_date: new Date("2023-07-07"),
    content:
      "模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。",
  },
  {
    editor_id: "admin1",
    project_id: "3",
    question_id: "1",
    edit_date: new Date("2023-07-07"),
    content:
      "模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。",
  },
  {
    editor_id: "admin1",
    project_id: "3",
    question_id: "2",
    edit_date: new Date("2023-07-07"),
    content:
      "模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。",
  },
  {
    editor_id: "admin1",
    project_id: "3",
    question_id: "3",
    edit_date: new Date("2023-07-07"),
    content:
      "模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。",
  },
];

export const Comment = [
  {
    user_id: "user1",
    admin_id: "adminuser",
    project_id: "1",
    comment_date: new Date("2023-07-07"),
    content:
      "コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。",
  },
  {
    user_id: "user1",
    admin_id: "adminuser",
    project_id: "2",
    comment_date: new Date("2023-07-07"),
    content:
      "コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入りますコメント内容が入ります。",
  },
  {
    user_id: "user1",
    admin_id: "adminuser",
    project_id: "3",
    comment_date: new Date("2023-07-07"),
    content:
      "コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入りますコメント内容が入ります。",
  },
];

export const User = [
  {
    id: "adminuser",
    name: "管理者太郎",
    image_url: "/icon/human_icon.png",
  },
];
