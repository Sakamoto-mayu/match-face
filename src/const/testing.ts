export const project = [
  {
    id: "1",
    name: "案件のタイトルが入ります",
    date: new Date("2023/07/01"),
    project_detail:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
  },
  {
    id: "2",
    name: "案件のタイトルが入ります",
    date: new Date("2023-07-02"),
    project_detail:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
  },
  {
    id: "3",
    name: "案件のタイトルが入ります",
    date: new Date("2023-07-03"),
    project_detail:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
  },
];

export const question = [
  {
    id: "1",
    project_id: "1",
    name: "質問のタイトルが入ります",
    content:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
    type: "writing",
  },
  {
    id: "2",
    project_id: "1",
    name: "質問のタイトルが入ります",
    content:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
    type: "select",
    choices: ["選択肢A", "選択肢B", "選択肢C", "選択肢D"],
  },
  {
    id: "3",
    project_id: "1",
    name: "質問のタイトルが入ります",
    content:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
    type: "writing",
  },
  {
    id: "4",
    project_id: "2",
    name: "質問のタイトルが入ります",
    content:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
    type: "writing",
  },
  {
    id: "5",
    project_id: "2",
    name: "質問のタイトルが入ります",
    content:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
    type: "writing",
  },
  {
    id: "6",
    project_id: "2",
    name: "質問のタイトルが入ります",
    content:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
    type: "select",
    choices: ["選択肢A", "選択肢B", "選択肢C", "選択肢D"],
  },
  {
    id: "7",
    project_id: "3",
    name: "質問のタイトルが入ります",
    content:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
    type: "writing",
  },
  {
    id: "8",
    project_id: "3",
    name: "質問のタイトルが入ります",
    content:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
    type: "select",
    choices: ["選択肢A", "選択肢B", "選択肢C", "選択肢D"],
  },
  {
    id: "9",
    project_id: "3",
    name: "質問のタイトルが入ります",
    content:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
    type: "writing",
  },
];
