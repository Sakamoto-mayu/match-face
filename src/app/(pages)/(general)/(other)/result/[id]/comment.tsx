import TextArea from "@/components/ui/TextArea";
import { Comment } from "@/const/result";
import { User } from "@/const/result";
import Image from "next/image";

type Props = {
  user_id: number;
  project_id: number;
};

export const CommentContent = ({ user_id, project_id }: Props) => {
  const currentComment = Comment.filter(
    (comment) => comment.user_id === user_id && project_id === project_id
  );
  const commentUser = User.filter(
    (user) => user.id === currentComment[0].admin_id
  );

  return (
    <>
      <h3 className="text-xl mb-3">{commentUser[0].name}からのコメント</h3>
      <div className="border border-black rounded-md p-3 flex justify-between ">
        <div className="w-1/6 flex justify-center items-center">
          <Image
            src={commentUser[0].image_url}
            alt="コメントしたユーザーのアイコン"
            width={70}
            height={70}
            priority
          />
        </div>
        <div id="comment" className="w-5/6">
          {currentComment[0].content}
        </div>
      </div>
    </>
  );
};
