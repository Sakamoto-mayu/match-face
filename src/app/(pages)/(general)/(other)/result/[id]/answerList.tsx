import { question } from "@/const/testing";
import { Answer } from "@/const/result";
import ModelAnswerContent from "@/components/pages/general/result/ModelAnswer";

type Props = {
  project_id: number;
};

export const AnswerList = ({ project_id }: Props) => {
  const currentQuestion = question.filter(
    (question) => question.project_id === project_id
  );

  const currentAnswer = Answer.filter(
    (answer) => answer.project_id === project_id
  );

  return (
    <>
      {currentQuestion.map((question, index) => (
        <div key={index} className="py-5">
          <h3 className="text-xl">
            Q{index + 1}:{question.name}
          </h3>
          <p id={`content-${index + 1}`} className="mt-2 mb-4">
            {question.content}
          </p>
          {question.type === "writing" ? (
            <div
              id={`answer-${index + 1}`}
              className="border border-gray p-3 mt-2 rounded-md shadow-md"
            >
              <h3 className="mb-2 text-xl">あなたの回答</h3>
              {currentAnswer[index].content}
            </div>
          ) : (
            <>
              <ul>
                {question.choices?.map((choice, index) => (
                  <li key={index}>・{choice}</li>
                ))}
              </ul>
              <div
                id={`answer-${index + 1}`}
                className="border border-gray p-3 mt-5 rounded-md shadow-md"
              >
                <h3 className="mb-2 text-xl">あなたの回答</h3>
                {currentAnswer[index].content}
              </div>
            </>
          )}
          <ModelAnswerContent
            project_id={project_id}
            question_id={question.id}
          />
        </div>
      ))}
    </>
  );
};
