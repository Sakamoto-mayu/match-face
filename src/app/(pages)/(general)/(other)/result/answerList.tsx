import { question } from "@/const/testing";
import { Answer } from "@/const/result";
import ModelAnswerContent from "./modelAnswer";

type Props = {
  project_id: number;
};

export const AnswerList = ({ project_id }: Props) => {
  const currentQuestion = question.filter(
    (question) => question.project_id === project_id
  );

  const currentAnswer = Answer.filter(
    (answer) => answer.project_id === project_id
  );

  return (
    <>
      {currentQuestion.map((question, index) => (
        <div key={index} className="py-5">
          <h3 className="text-xl">
            Q{index + 1}：{question.name}
          </h3>
          <p className="mt-2 mb-4">{question.content}</p>
          {question.type === "writing" ? (
            <>
              <div className="border border-gray p-3 mt-2 rounded-md shadow-md">
                <h3 className="mb-2 text-xl">あなたの回答</h3>
                {currentAnswer[index].content}
              </div>
            </>
          ) : (
            <>
              {question.choices?.map((choice, index) => (
                <div key={index}>
                  <input
                    id={`choice${index}`}
                    type="radio"
                    value={choice}
                    name={question.name}
                  />
                  <label htmlFor={`choice${index}`}>{choice}</label>
                </div>
              ))}
            </>
          )}
          <ModelAnswerContent
            project_id={project_id}
            question_id={question.id}
          />
        </div>
      ))}
    </>
  );
};
