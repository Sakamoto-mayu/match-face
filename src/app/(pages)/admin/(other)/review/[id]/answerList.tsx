import { question } from "@/const/testing";
import { Answer } from "@/const/review";
import { User } from "@/const/review";
import ModelAnswerContent from "./modelAnswer";

type Props = {
  project_id: string;
  user_id: string;
};

export const AnswerList = ({ user_id, project_id }: Props) => {
  const currentQuestion = question.filter(
    (question) => question.project_id === project_id
  );

  const currentAnswer = Answer.filter(
    (answer) => answer.project_id === project_id
  );

  const answerUser = User.filter((user) => user.id === user_id);

  return (
    <>
      {currentQuestion.map((question, index) => (
        <div key={index} className="py-5">
          <h3 className="text-xl">
            Q{index + 1}：{question.name}
          </h3>
          <p className="mt-2 mb-4">{question.content}</p>
          {question.type === "writing" ? (
            <>
              <div className="border border-gray p-3 mt-2 rounded-md shadow-md">
                <h3 className="mb-2 text-xl">{answerUser[0].name}の回答</h3>
                {currentAnswer[index].content}
              </div>
            </>
          ) : (
            <>
              <ul>
                {question.choices?.map((choice, index) => (
                  <li key={index}>・{choice}</li>
                ))}
              </ul>
              <div className="border border-gray p-3 mt-5 rounded-md shadow-md">
                <h3 className="mb-2 text-xl">{answerUser[0].name}の回答</h3>
                {currentAnswer[index].content}
              </div>
            </>
          )}
          <ModelAnswerContent
            project_id={project_id}
            question_id={question.id}
          />
        </div>
      ))}
    </>
  );
};
