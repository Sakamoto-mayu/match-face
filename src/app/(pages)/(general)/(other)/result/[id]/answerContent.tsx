"use client";

import { project } from "@/const/testing";
import { AnswerList } from "./answerList";
import { CommentContent } from "./comment";
import OrangeButton from "@/components/ui/button/OrangeButton";
import { Answer } from "@/const/result";
import { useRouter } from "next/navigation";

type Props = {
  user_id: number;
  project_id: number;
};

export const AnswerContent = ({ user_id, project_id }: Props) => {
  const router = useRouter();
  const currentProject = project.filter(
    (project) => project.id === Number(project_id)
  );
  const currentAnswer = Answer.filter(
    (answer) => answer.project_id === Number(project_id)
  );

  return (
    <div className="w-1/2 py-5">
      <div className="flex justify-between items-center my-3">
        <h1 className="text-2xl">回答の確認</h1>
        {/*現状のデータでは4以降の解答データを呼び出そうとするとエラーになります */}
        <p>回答日時:{currentAnswer[0].answer_date.toLocaleDateString()}</p>
      </div>
      <hr className="border border-black" />
      <h2 className="text-xl my-2">案件名：{currentProject[0].name}</h2>
      <p className="text-lg">{currentProject[0].project_detail}</p>
      <AnswerList project_id={project_id} />
      <CommentContent user_id={user_id} project_id={project_id} />
      <div className="flex justify-around my-20">
        <OrangeButton
          className="border rounded-none"
          label="質問一覧へ"
          onClick={() => router.push("/questions")}
        />
        <OrangeButton
          className="border rounded-none"
          label="回答履歴一覧へ"
          onClick={() => router.push("/histories")}
        />
      </div>
    </div>
  );
};
