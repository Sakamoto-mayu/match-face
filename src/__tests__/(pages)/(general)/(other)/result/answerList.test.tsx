import { AnswerList } from "@/app/(pages)/(general)/(other)/result/[id]/answerList";
import { render, screen } from "@testing-library/react";
import { question } from "@/const/testing";
import { Answer } from "@/const/result";
import "@testing-library/jest-dom";

const testData = {
  project_id: "1",
};

const currentQuestion = question.filter(
  (question) => question.project_id === testData.project_id
);

const currentAnswer = Answer.filter(
  (answer) => answer.project_id === testData.project_id
);

describe("AnswerListコンポーネントのテスト", () => {
  test("回答内容が表示される", () => {
    render(<AnswerList project_id={testData.project_id} />);

    // 質問のタイトル・内容が一致する
    for (let i = 0; i <= 2; i++) {
      const question_title = screen.getByRole("heading", {
        level: 3,
        name: `Q${i + 1}:${currentQuestion[i].name}`,
      });
      const question_content = document.querySelector(`#content-${i + 1}`);

      expect(question_title).toBeInTheDocument();
      expect(question_content).toHaveTextContent(currentQuestion[i].content);
    }

    // 回答の表示数と質問の数が一致する
    const answer_title = screen.getAllByRole("heading", {
      level: 3,
      name: "あなたの回答",
    });
    expect(answer_title).toHaveLength(3);

    // 回答内容が一致する
    for (let i = 0; i <= 2; i++) {
      const answer_content = document.querySelector(`#answer-${i + 1}`);
      expect(answer_content).toHaveTextContent(currentAnswer[i].content);
    }
  });

  test("スナップショットテスト", () => {
    const contents = render(<AnswerList project_id={testData.project_id} />);
    expect(contents).toMatchSnapshot();
  });
});
