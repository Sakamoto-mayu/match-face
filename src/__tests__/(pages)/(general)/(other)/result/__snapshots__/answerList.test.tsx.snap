// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnswerListコンポーネントのテスト スナップショットテスト 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="py-5"
      >
        <h3
          class="text-xl"
        >
          Q
          1
          :
          質問のタイトルが入ります
        </h3>
        <p
          class="mt-2 mb-4"
          id="content-1"
        >
          テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト
        </p>
        <div
          class="border border-gray p-3 mt-2 rounded-md shadow-md"
          id="answer-1"
        >
          <h3
            class="mb-2 text-xl"
          >
            あなたの回答
          </h3>
          回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。
        </div>
        <div
          class="my-7 p-3 rounded-md bg-light-gray"
          id="modelAnswer"
        >
          <h3
            class="text-xl mb-2 "
          >
            模範解答
          </h3>
          模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。
        </div>
      </div>
      <div
        class="py-5"
      >
        <h3
          class="text-xl"
        >
          Q
          2
          :
          質問のタイトルが入ります
        </h3>
        <p
          class="mt-2 mb-4"
          id="content-2"
        >
          テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト
        </p>
        <ul>
          <li>
            ・
            選択肢A
          </li>
          <li>
            ・
            選択肢B
          </li>
          <li>
            ・
            選択肢C
          </li>
          <li>
            ・
            選択肢D
          </li>
        </ul>
        <div
          class="border border-gray p-3 mt-5 rounded-md shadow-md"
          id="answer-2"
        >
          <h3
            class="mb-2 text-xl"
          >
            あなたの回答
          </h3>
          選択肢A
        </div>
        <div
          class="my-7 p-3 rounded-md bg-light-gray"
          id="modelAnswer"
        >
          <h3
            class="text-xl mb-2 "
          >
            模範解答
          </h3>
          模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。
        </div>
      </div>
      <div
        class="py-5"
      >
        <h3
          class="text-xl"
        >
          Q
          3
          :
          質問のタイトルが入ります
        </h3>
        <p
          class="mt-2 mb-4"
          id="content-3"
        >
          テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト
        </p>
        <div
          class="border border-gray p-3 mt-2 rounded-md shadow-md"
          id="answer-3"
        >
          <h3
            class="mb-2 text-xl"
          >
            あなたの回答
          </h3>
          回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。
        </div>
        <div
          class="my-7 p-3 rounded-md bg-light-gray"
          id="modelAnswer"
        >
          <h3
            class="text-xl mb-2 "
          >
            模範解答
          </h3>
          模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="py-5"
    >
      <h3
        class="text-xl"
      >
        Q
        1
        :
        質問のタイトルが入ります
      </h3>
      <p
        class="mt-2 mb-4"
        id="content-1"
      >
        テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト
      </p>
      <div
        class="border border-gray p-3 mt-2 rounded-md shadow-md"
        id="answer-1"
      >
        <h3
          class="mb-2 text-xl"
        >
          あなたの回答
        </h3>
        回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。
      </div>
      <div
        class="my-7 p-3 rounded-md bg-light-gray"
        id="modelAnswer"
      >
        <h3
          class="text-xl mb-2 "
        >
          模範解答
        </h3>
        模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。
      </div>
    </div>
    <div
      class="py-5"
    >
      <h3
        class="text-xl"
      >
        Q
        2
        :
        質問のタイトルが入ります
      </h3>
      <p
        class="mt-2 mb-4"
        id="content-2"
      >
        テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト
      </p>
      <ul>
        <li>
          ・
          選択肢A
        </li>
        <li>
          ・
          選択肢B
        </li>
        <li>
          ・
          選択肢C
        </li>
        <li>
          ・
          選択肢D
        </li>
      </ul>
      <div
        class="border border-gray p-3 mt-5 rounded-md shadow-md"
        id="answer-2"
      >
        <h3
          class="mb-2 text-xl"
        >
          あなたの回答
        </h3>
        選択肢A
      </div>
      <div
        class="my-7 p-3 rounded-md bg-light-gray"
        id="modelAnswer"
      >
        <h3
          class="text-xl mb-2 "
        >
          模範解答
        </h3>
        模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。
      </div>
    </div>
    <div
      class="py-5"
    >
      <h3
        class="text-xl"
      >
        Q
        3
        :
        質問のタイトルが入ります
      </h3>
      <p
        class="mt-2 mb-4"
        id="content-3"
      >
        テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト
      </p>
      <div
        class="border border-gray p-3 mt-2 rounded-md shadow-md"
        id="answer-3"
      >
        <h3
          class="mb-2 text-xl"
        >
          あなたの回答
        </h3>
        回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。
      </div>
      <div
        class="my-7 p-3 rounded-md bg-light-gray"
        id="modelAnswer"
      >
        <h3
          class="text-xl mb-2 "
        >
          模範解答
        </h3>
        模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
